### PR TITLE
Implements CameraAPI

### PR DESCRIPTION
**Pull request by kaxyum**
## Description

Implements CameraAPI for the new Camera System to PocketMine-MP

## Changes
adds packet sending when the player joins the server in ``pocketmine\player\Player``
new classes added : ``pocketmine\camera\VanillaCameraPresets``, ``pocketmine\camera\instruction\CameraInstruction``, ``pocketmine\camera\instruction\ClearCameraInstruction``, ``pocketmine\camera\instruction\FadeCameraInstruction``, ``pocketmine\camera\instruction\SetCameraInstruction``

## Video

https://github.com/pmmp/PocketMine-MP/assets/98211455/6aace09d-914a-4aea-ac99-aab92119fb99

### Set Camera Instruction
```php
// only the argument preset is compulsory
$setCameraInstruction = new SetCameraInstruction();
$setCameraInstruction->setPreset(CameraPresets::FREE()); //CameraPresets::FIRST_PERSON(), CameraPresets::THIRD_PERSON(), CameraPresets::THIRD_PERSON_FRONT()
$setCameraInstruction->setEase(CameraSetInstructionEaseType::LINEAR, 1);
$setCameraInstruction->setCameraPostion(new Vector3(100, 100, 100));
$setCameraInstruction->setRotation(0, 0);
$setCameraInstruction->setFacingPosition(new Vector3(0, 0, 0));
$player->sendCameraInstruction($setCameraInstruction);
```
### Fade Camera Instruction
```php
// no argument is compulsory
$fadeCameraInstruction = new FadeCameraInstruction();
$fadeCameraInstruction->setTime(1, 5, 1);
$fadeCameraInstruction->setColor(225, 225, 225);
$player->sendCameraInstruction($fadeCameraInstruction);
```

### Clear Camera Instruction
```php
// no argument is compulsory
$clearCameraInstruction = new ClearCameraInstruction();
$clearCameraInstruction->setClear(true);
$player->sendCameraInstruction($clearCameraInstruction);
```

## Tests

I tested this PR by doing the following:
- [ ] Writing PHPUnit tests (commit these in the `tests/phpunit` folder)
- [x] Playtesting using a Minecraft client (provide screenshots or a video)
- [x] Writing a test plugin (provide the code and sample output)
- [ ] Other (provide details)

### CODE REVIEW

**Comment by ShockedPlot7560** [comment]
Should target `next-minor`
The namespace seems more logical to be inside the `player/`

File: src/camera/VanillaCameraPresets.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ * @generate-registry-docblock
+ *
+ * @method static CameraPreset FREE()
+ * @method static CameraPreset FIRST_PERSON()
+ * @method static CameraPreset THIRD_PERSON()
+ * @method static CameraPreset THIRD_PERSON_FRONT()
+ */
+
+final class VanillaCameraPresets{
+	use RegistryTrait;
```
Comment (ShockedPlot7560) [inline src/camera/VanillaCameraPresets.php:42]: A native enum could be used instead, no?

File: src/camera/instruction/ClearCameraInstruction.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+	public function setClear(bool $clear) : void
+	{
+		$this->clear = $clear;
+	}
+
+	public function send(Player $player) : void
+	{
+		$player->getNetworkSession()->sendDataPacket(CameraInstructionPacket::create(null, $this->clear, null));
+	}
```
Comment (ShockedPlot7560) [inline src/camera/instruction/ClearCameraInstruction.php:41]: Would it reduce code coupling to move this function into the player?

File: src/player/Player.php [hunk 3]
```diff
@@ -2710,4 +2718,14 @@ public function onChunkUnloaded(int $chunkX, int $chunkZ, Chunk $chunk) : void{
 			$this->unloadChunk($chunkX, $chunkZ);
 		}
 	}
+
+	public function sendCameraInstruction(CameraInstruction $cameraInstruction) : void{
+		if($cameraInstruction instanceof SetCameraInstruction){
```
Comment (SmallkingDev) [inline src/player/Player.php:2723]: There's no need to complicate this. Just leave it at that:

```php
$this->getNetworkSession()->sendDataPacket(CameraInstructionPacket::create($cameraInstruction, null, null));
```

**Comment by Gewinum** [comment]
phpstan is so aggressive

**Pull request closed without merging**